### ISSUE TITLE
run_udf: Unclear why two data types are defined

### ISSUE THREAD

**Issue by m-mohr** [Open-EO/openeo-processes#515]
**Process ID:** run_udf

**Describe the issue:**
Run UDF has the following definition for `data`:

```json
        {
            "name": "data",
            "description": "The data to be passed to the UDF.",
            "schema": [
                {
                    "title": "Array",
                    "type": "array",
                    "minItems": 1,
                    "items": {
                        "description": "Any data type."
                    }
                },
                {
                    "title": "Single Value",
                    "description": "A single value of any data type."
                }
            ]
        }
```

Why does it explicitly define an array type? The way the second schema is defined, both apply for an array. So it's equivalent to:

```json
        {
            "name": "data",
            "description": "The data to be passed to the UDF.",
            "schema": {
                "title": "Single Value",
                "description": "A single value of any data type."
            }
        }
```

Can someone remember why this is the way it is? @soxofaan maybe?

**Proposed solution:**
Simplify the schema?

**Comment by m-mohr** [comment]
Git blame shows that this was changed 4 years ago (1e4c730ad8495a4ab75bf854e0bed23e0f75e531) from two schemas (raster-cube and array) to two schemas (array and any). I believe this was an oversight and shall be just one schema that allows any type.

**Comment by soxofaan** [comment]
> changed 2021 (https://github.com/Open-EO/openeo-processes/commit/6ec1848b1acf37fdacb8467de86a3e63c1325657) from two schemas (raster-cube and array) to two schemas (array and any).

you mean from three schemas (raster-cube, array and any) to two schemas (array and any)


regardless, I have no idea why it's like this or if there are important reasons not to simplify as proposed.
The only reason I can think of is to make things a bit more self-documenting, but I guess it's then just better to improve descriptions/dcos than keeping confusing  schema constructs

**Comment by m-mohr** [comment]
Yes, I agree... I guess it was meant as a way to somehow recommend that UDFs are used in callbacks.
